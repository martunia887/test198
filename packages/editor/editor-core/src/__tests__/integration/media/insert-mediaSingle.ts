import { BrowserTestCase } from '@atlaskit/webdriver-runner/runner';
import {
  editable,
  expectMatchDocument,
  insertMedia,
  fullpage,
} from '../_helpers';
import {
  mountEditor,
  goToEditorTestingExample,
} from '../../__helpers/testing-example-helpers';

BrowserTestCase(
  'insert-mediaSingle.ts: Inserts a media single on fullpage',
  { skip: ['edge', 'ie', 'safari'] },
  async (client: any, testName: string) => {
    const page = await goToEditorTestingExample(client);
    await mountEditor(page, {
      appearance: fullpage.appearance,
      media: {
        allowMediaSingle: true,
        allowMediaGroup: true,
      },
    });

    // type some text
    await page.click(editable);
    await page.type(editable, 'some text');

    // now we can insert media as necessary
    await insertMedia(page);

    expect(await page.isVisible('.media-single')).toBe(true);

    await expectMatchDocument(page, testName);
  },
);
