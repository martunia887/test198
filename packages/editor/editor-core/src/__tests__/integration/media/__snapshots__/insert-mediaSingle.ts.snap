// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`insert-mediaSingle.ts: Inserts a media single on fullpage 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": "some text",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "attrs": Object {
        "layout": "center",
      },
      "content": Array [
        Object {
          "attrs": Object {
            "__fileName": "example.png",
            "collection": "MediaServicesSample",
            "height": 4096,
            "id": "1234-5678-abcd-efgh",
            "type": "file",
            "width": 4096,
          },
          "type": "media",
        },
      ],
      "type": "mediaSingle",
    },
    Object {
      "type": "paragraph",
    },
  ],
  "type": "doc",
  "version": 1,
}
`;
